src/main/java/org/molgenis/vcf/decisiontree/filter/VcfRecord.java

```diff
@@ -60,7 +69,53 @@ public Object getValue(Field field, int alleleIndex) {
     return value;
   }
 
-  private Object getCommonValue(Field field, int alleleIndex) {
+  private Object getNestedValue(Field field, Allele allele) {
```

Comment:
Could use some unit tests (currently not covered by any test cases). There is a lot going on in the complex piece of code. Lets have a short discussion to see if we could make some changes to reduce complexity.